server/src/main/java/io/crate/types/RowType.java

```diff
@@ -176,7 +178,7 @@ public TypeSignature getTypeSignature() {
             var fieldType = fieldTypes.get(i);
             parameters.add(
                 new ParameterTypeSignature(
-                    fieldName,
+                    escapeTypeSignatureValues(fieldName),
```

Comment:
Could we move this into the `ParameterTypeSignature` constructor? I suspect it would be less error-prone there because it would be in one location instead of many.